node-manager/src/wallet.rs

```diff
@@ -57,22 +59,34 @@ impl MutinyWallet {
         }
     }
 
-    pub async fn sync(&self) -> Result<(), bdk::Error> {
+    pub async fn sync(&self) -> Result<(), MutinyError> {
         let wallet = self.wallet.lock().await;
 
-        wallet.sync(&self.blockchain, SyncOptions::default()).await
+        Ok(wallet
+            .sync(&self.blockchain, SyncOptions::default())
+            .await?)
     }
 
     pub async fn send(
         &self,
         destination_address: String,
         amount: u64,
         fee_rate: Option<f32>,
-    ) -> Result<bitcoin::Txid, bdk::Error> {
+    ) -> Result<bitcoin::Txid, MutinyError> {
         let wallet = self.wallet.lock().await;
 
-        let send_to = Address::from_str(&destination_address)
-            .map_err(|e| bdk::Error::Generic(e.to_string()))?;
+        // TODO: would like to be able to convert from Bitcoin lib errors directly to MutinyError somehow
+        // Like this doesn't seem to do anything:
+        // #[derive(Error, Debug)]
+        // pub enum MutinyBitcoinError {
+        //     #[error("Failed to use browser storage")]
+        //     AddressError {
+        //         #[from]
+        //         source: bitcoin::util::address::Error,
+        //     },
+        // }
+        let send_to =
+            Address::from_str(&destination_address).with_context(|| "Address parse error")?;
```

Comment:
what's `with_context`?

Reply:
here's how they explain it in the readme:

```
Attach context to help the person troubleshooting the error understand where things went wrong. A low-level error like "No such file or directory" can be annoying to debug without more context about what higher level step the application was in the middle of.
```